src/core/thread/mle.cpp

```diff
@@ -2315,14 +2363,19 @@ ThreadError Mle::HandleAnnounce(const Message &aMessage, const Ip6::MessageInfo
 
     localTimestamp = mNetif.GetActiveDataset().GetNetwork().GetTimestamp();
```

Comment:
When attached to a network, we should be using the version disseminated by the network.  When not attached to the network, we should be using the locally stored version.  There's a couple other instances of this.  I think it's better to clean this interaction up in a separate PR.